go-controller/pkg/crd/userdefinednetwork/v1/cudn.go

```diff
@@ -26,6 +26,7 @@ type ClusterUserDefinedNetworkSpec struct {
 	// NamespaceSelector Label selector for which namespace network should be available for.
 	// +kubebuilder:validation:Required
 	// +required
+	// +kubebuilder:validation:XValidation:rule="self.matchLabels.size() > 0 || self.matchExpressions.size() > 0", message="namespace selector must have at least one rule"
```

Comment:
it may be better to add has() check, like
`has(self.matchLabels) && self.matchLabels.size() > 0 || has(self.matchExpressions) && self.matchExpressions.size() > 0` so that the error message is more clean.
For example, if you apply the following selector
```
namespaceSelector:
    matchExpressions: []
```
you will get this error
`spec.namespaceSelector: Invalid value: "object": no such key: matchLabels evaluating rule: namespace selector must have at least one rule` which looks like matchLabels is required, but it is not the case, with the suggested rule you will get
`spec.namespaceSelector: Invalid value: "object": namespace selector must have at least one rule`

Reply:
Done